### DISCUSSION TITLE
Extract PDF viewer to a separate app

### DISCUSSION THREAD

**Discussion by tswistak** [FossifyOrg/General-Discussion#486] [Ideas]
I think it would be a good idea to separate the PDF viewer from the File Manager app.

- If someone wants to use it, it can be installed as a separate app. Thanks to this, File Manager will again weigh around 8MB, not 33MB.
- It will allow people to use a PDF viewer without File Manager. For example, I don't use File Manager because I can't do all my use cases with it (sending files via LAN with SMB), but I'd like to have a nice FOSS PDF viewer.
- Potential vulnerabilities in PDF viewer won't block the whole File Manager from app repositories, like it was before. Especially, that it's based on Tibor's fork of an unmaintained library. The question is how long we can be sure there won't be any new vulnerabilities, that will never get fixed.

As I remember, there were discussions about it in SMT and people wanted a separate app, but Tibor wanted it to be integrated into File Manager.

**Comment by naveensingh** [comment]
Yes, I have been thinking about lowering app size across all apps.

> I think it would be a good idea to separate the PDF viewer from the File Manager app

It _is_ a good idea to separate the PDF viewer and the File Manager app.

> As I remember, there were discussions about it in SMT and people wanted a separate app, but Tibor wanted it to be integrated into File Manager.

Yep, I know.

**Comment by inson1** [comment]
I wouldnt call it pdf viewer, we could add support for other formats too

**Reply by naveensingh** [reply]
That's why I'm going for the documents icon. It'll still be called PDF Viewer because that's what it'll be. A basic PDF Viewer with or without a main page. It'll be later renamed to Fossify Documents if and when we add support for more formats.

**Comment by gbdomubpkm** [comment]
Good idea indeed to have a separate app for reading pdf. 👍

**Comment by realeatham** [comment]
If this is made, it would be good if it could also edit and display text , html, and markdown. Maybe even an option to force display a file in another format could be good

**Comment by naveensingh** [comment]
Closing this as it's already done, I'm working on the PDF Viewer app and I hope to release it soon (ish).

**Reply by matj1** [reply]
Is the standalone app already released? And where could I find it?

**Reply by naveensingh** [reply]
Not yet.

https://github.com/FossifyOrg/Documents

**Reply by zedxxx** [reply]
Any updates on the PDF Viewer? Looking at the Documents repository, it seems that development has completely stalled. It's disappointing to see such a useful feature get abandoned.

**Reply by naveensingh** [reply]
It was supposed to be released this month, but sometimes I'm extra slow.

**Reply by tswistak** [reply]
Any news? October is already long gone...

**Reply by BrandQc** [reply]
Any news?!
